### ISSUE TITLE
Unable to build "With C++" demo

### ISSUE THREAD

**Issue by Pseudopode** [schellingb/ClangWasm#1]
I can successfully build all the demo, but not the third one ("With C++"), on Windows 10.
Everything is set up as asked in the tutorial. The other examples can be built properly.
However when I try to build the third one, here's the output from the console :

`Copying loader.js to Release-wasm/loader.js ...`
`main.cpp`
`Linking Release-wasm/output.wasm ...`
`wasm-ld: error: Release-wasm/main.o: undefined symbol: std::__2::cout`
`wasm-ld: error: Release-wasm/main.o: undefined symbol: std::__2::ctype<char>::id`
`Makefile:106: recipe for target 'Release-wasm/output.wasm' failed`
`make: *** [Release-wasm/output.wasm] Error 1`

I tried a diff with the makefile from the 7th tutorial, but couldn't build it anyway.

Is there something needed to let the linker link against std::cout ?

**Comment by schellingb** [comment]
Hi there
Thanks so much for testing these and finding this mistake.

One of the last sentences I wrote for the blog post was:

> If you’re wondering why it’s still using printf and not std::cout, I have disabled streams and locale on purpose because it makes the output 180 KB instead 20 KB.

So at the last moment before uploading I was trying out std::cout and somehow that (invalid) line of code was left there in the .cpp file.

I removed the line for now and it should compile.

I was thinking of adding another demo that builds the full C++ standard library with streams support for people who use them. It is a bit silly as std::cout needs streams, streams need locale, locale then need timezone, mmap and thread support which needs to be faked or emulated for wasm. All that just for "modern C++" printf...

Thanks again for reporting!
